packages/react/src/composite/list/CompositeList.tsx

```diff
@@ -102,6 +102,19 @@ export function CompositeList<Metadata>(props: CompositeList.Props<Metadata>) {
     onMapChange?.(sortedMap);
   }, [onMapChange, sortedMap, elementsRef, labelsRef, mapTick]);
 
+  const cleanupRefs = useEventCallback(() => {
+    elementsRef.current.length = 0;
+    if (labelsRef) {
+      labelsRef.current.length = 0;
+    }
+  });
+
+  useIsoLayoutEffect(() => {
+    return () => {
+      cleanupRefs();
+    };
+  }, [cleanupRefs]);
```

Comment:
The changes in CompoboxList revealed a bug where refs are not cleaned up on unmount, keeps holding all references to the elements. 
This caused the test fail in React 18 Strict Mode, because `listRef` is not empty on the second+ time on open and `waitForListPopulated` in `useListNavigation` assumes it is ready to navigate as soon as it opens.

This was not an issue in React 19 because `IndexGuessBehavior.GuessFromOrder` works more accurate.